edge-middleware/bot-protection-datadome/lib/datadome.ts

```diff
@@ -1,82 +1,98 @@
 import { NextRequest, NextResponse } from 'next/server'
 
-const DATADOME_TIMEOUT = 500
+const DATADOME_TIMEOUT = parseInt(process.env.DATADOME_TIMEOUT ?? "300")
+const DATADOME_ENDPOINT = process.env.DATADOME_ENDPOINT ?? "https://api.datadome.co"
 const DATADOME_URI_REGEX_EXCLUSION =
-  /\.(avi|flv|mka|mkv|mov|mp4|mpeg|mpg|mp3|flac|ogg|ogm|opus|wav|webm|webp|bmp|gif|ico|jpeg|jpg|png|svg|svgz|swf|eot|otf|ttf|woff|woff2|css|less|js)$/i
+  /\.(avi|flv|mka|mkv|mov|mp4|mpeg|mpg|mp3|flac|ogg|ogm|opus|wav|webm|webp|bmp|gif|ico|jpeg|jpg|png|svg|svgz|swf|eot|otf|ttf|woff|woff2|css|less|js|map)$/i
 
 export default async function datadome(req: NextRequest) {
   const { pathname } = req.nextUrl
 
   if (DATADOME_URI_REGEX_EXCLUSION.test(pathname)) {
     return
   }
-
+  let { clientId, cookiesLength } = getCookieData(req.cookies)
   const requestData = {
-    Key: process.env.DATADOME_SERVER_KEY,
-    RequestModuleName: 'Next.js',
-    ModuleVersion: '0.1',
-    ServerName: 'vercel',
+    Key: process.env.DATADOME_SERVER_SIDE_KEY,
     // this should be `x-real-ip` but it doesn't currently work on Edge Middleware
+    // localhost won't likely be blocked by Datadome unless you use your real IP
+    // IP: 'YOUR IP',
     IP: req.headers.get('x-forwarded-for')
       ? req.headers.get('x-forwarded-for')!.split(',')[0]
       : '127.0.0.1',
-    // localhost won't likely be blocked by Datadome unless you use your real IP
-    // IP: 'YOUR IP',
-    Port: 0,
-    TimeRequest: new Date().getTime() * 1000,
-    Protocol: req.headers.get('x-forwarded-proto'),
-    Method: req.method,
-    ServerHostname: req.headers.get('host'),
-    Request: pathname + encode(Object.fromEntries(req.nextUrl.searchParams)),
-    HeadersList: getHeadersList(req),
-    Host: req.headers.get('host'),
-    UserAgent: req.headers.get('user-agent'),
-    Referer: req.headers.get('referer'),
-    // Make sure Datadome always returns a JSON response in case of a 403
-    Accept: 'application/json',
+    RequestModuleName: 'Next.js',
+    ModuleVersion: '1.0.0',
```

Comment:
Headers are not truncated and it is in the definition of done for a module:https://www.notion.so/datadome/Definition-of-DONE-for-a-module-integration-c87abcd8d16f4359a6a79b4d75d6faa0
Should we do it in this version?

Reply:
Thanks, header are now truncated